File: Presentation/Erwan/static/fibonacci.c
#include <stdio.h>
int fibonacci_rec(int n){


    int retval;

    if (n < 0){
        printf("%s", "Erreur: n ne doit pas etre negatif");
        retval =  -1;
        return retval;

    }
    if (n == 0){
        retval = 0;
        return retval;

    }    else if (n == 1){
        retval = 1;
        return retval;

    }    else {
        retval = retval;
    }

    retval = fibonacci_rec(n - 1) + fibonacci_rec(n - 2);


    return retval;
}
// fibonacci
void main(void){
    int n, result_rec;
    n = 10;
    // Appel de la fonction recursive
    result_rec = fibonacci_rec(n);
    printf("%s%d%s%d", "Fibonacci (recursif) de ", n, " est ", result_rec);



}

File: Presentation/Erwan/static/fibonacci.f90
program fibonacci
    integer :: n, result_rec
    n = 10
    ! Appel de la fonction recursive
    result_rec = fibonacci_rec(n)
    print *, "Fibonacci (recursif) de ", n, " est ", result_rec

contains
    recursive function fibonacci_rec(n) result(retval)
        integer, intent(in) :: n
        integer :: retval

        if (n < 0) then
            print *, "Erreur: n ne doit pas etre negatif"
            retval = -1
            return
        end if
        if (n == 0) then
            retval = 0
            return
        else if (n == 1) then
            retval = 1
            return
        else
            retval = fibonacci_rec(n - 1) + fibonacci_rec(n - 2)
        end if

    end function fibonacci_rec
end program fibonacci